### PR TITLE
refactor: batch control interfaces in BFD

### PR DESCRIPTION
**Pull request by cyw233**
### Description of PR

Refactor the interfaces toggling in BFD static route to batch toggling to get a better performance.

Summary:
Fixes # (issue) Microsoft ADO 29473566

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [x] 202405

#### What is the motivation for this PR?
Currently, we are toggling the interfaces in `bfd/test_bfd_static_route.py` in a loop, causing long running time. We wanted to batch toggle the interfaces to improve the overall performance. 

#### How did you do it?
Convert the `for` loop toggling into a batch toggling and batch waiting.

#### How did you verify/test it?
I ran the updated code in T2 chassis and can confirm it passed.

### CODE REVIEW

File: tests/bfd/bfd_helpers.py [hunk 1]
```diff
@@ -158,41 +158,61 @@ def verify_static_route(
...
+            logger.info("Target state for interface {} is {}. Command: {}".format(
+                interface,
+                target_state,
+                exec_cmd,
+            ))
+        else:
+            raise ValueError("Invalid action specified for interface {}".format(interface))
+
+    if cmds:
+        dut.shell_cmds(cmds=cmds)
```
Comment (yejianquan) [inline tests/bfd/bfd_helpers.py:192]: I did a quick check on shell_cmds source code,
it actually runs the commands one by one: https://github.com/sonic-net/sonic-mgmt/blob/331254335e7455aa148f6d8c67d921e614afa4e4/ansible/library/shell_cmds.py#L137
So how much time we can expect to reduce with this refinement approach?

And how about we use multi-thread to run the interface commands? This way the commands will be executed in parallel

Comment (cyw233) [reply tests/bfd/bfd_helpers.py:192]: We are seeing 20 seconds improvement on this as we only run one `dut.show_interface()` before the interfaces control command with this implementation. You are right, we can further improve this by running the startup/shutdown interfaces commands in parallel with mult-threading. Let me make some changes. Thanks!

Comment (cyw233) [reply tests/bfd/bfd_helpers.py]: Hey @yejianquan, we are seeing another 45 seconds improvement after using multithreading to run those commands. So, in total we are seeing 65 seconds improvement for a single batch👍

**Review by yejianquan — APPROVED** [review]
LGTM

**Pull request merged**